boards/arm/apollo4p_blue_kxr_evb/apollo4p_blue_kxr_evb.dts

```diff
@@ -86,6 +86,23 @@
 	status = "okay";
 };
 
+&iom4 {
+	compatible = "ambiq,spi";
+	pinctrl-0 = <&spi4_default>;
+	pinctrl-names = "default";
+	cs-gpios = <&gpio32_63 22 (GPIO_ACTIVE_LOW | GPIO_PULL_UP)>;
+	clock-frequency = <DT_FREQ_M(24)>;
+	status = "okay";
+
+	bt-hci@0 {
+		compatible = "ambiq,bt-hci-spi";
+		reg = <0>;
+		irq-gpios = <&gpio32_63 21 GPIO_ACTIVE_HIGH>;
+		reset-gpios = <&gpio32_63 23 GPIO_ACTIVE_LOW>;
+		clkreq-gpios = <&gpio32_63 20 GPIO_ACTIVE_HIGH>;
+	};
+};
```

Comment:
Are these wired internally in the chip? If that's the case I think it would make sense to have the nodes declared in `ambiq_apollo4p_blue.dtsi` so that you don't have to redefine the same things on every board using the chip, it's not like you can change it anyway. See `stm32wl.dtsi` with the `stm32wl-subghz-radio` node in the internal spi for example.

Reply:
@fabiobaltieri thanks for your following. Will have an independent PR to optimize this.